### PR TITLE
refactor: configure kubelet serving certificate rotation label at runtime

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

**What this PR does / why we need it**:

this PR moves the logic which adds the special `kubernetes.azure.com/kubelet-serving-ca=cluster` label out of agentbaker and directly into provisioning scripts - this is primarily to ensure that we preserve this functionality when using scriptless, and there is no agentbakersvc

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: staging/cse/windows/kubeletfunc.ps1 [hunk 1]
```diff
@@ -209,6 +209,24 @@ function Get-KubePackage {
...
+        [Parameter(Mandatory=$true)][string]
+        $KubeletNodeLabels,
+        [Parameter(Mandatory=$true)][string]
+        $Label
+    )
+
+    $labelList = $KubeletNodeLabels -split ","
+    foreach ($existingLabel in $labelList) {
+        if ($existingLabel -eq $Label) {
+            return $KubeletNodeLabels
```
Comment (AbelHu) [inline staging/cse/windows/kubeletfunc.ps1:223]: Add logs here and line 226 so we can know when the label is added for troubleshooting?

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 2]
```diff
@@ -423,21 +423,37 @@ getPrimaryNicIP() {
...
+        KUBELET_NODE_LABELS="${KUBELET_NODE_LABELS},${LABEL_STRING}"
+    else
+        # node labels shouldn't ever be empty, but we guard against it to be safe
+        KUBELET_NODE_LABELS=$LABEL_STRING
+    fi
+}
+
 # removes the specified LABEL_STRING (which should be in the form of 'label=value') from KUBELET_NODE_LABELS
-clearKubeletNodeLabel() {
+removeKubeletNodeLabel() {
```
Comment (lilypan26) [inline parts/linux/cloud-init/artifacts/cse_config.sh:443]: nit: move this and `addKubeletNodeLabel` to cse_helpers? Might be useful in future for other cases

**Comment by lilypan26** [comment]
is it safe to remove the equivalent logic from `baker.go` files? Or do we still need it there as well

**Comment by cameronmeissner** [comment]
> is it safe to remove the equivalent logic from `baker.go` files? Or do we still need it there as well

we need to keep it in there for now since we can't release everything at once for windows (they need to create a new script package before we can start calling new functions from within the primary CSE scripts)

I'll follow-up and remove that logic once windows has everything released

File: parts/linux/cloud-init/artifacts/cse_config.sh (hunk 2 above)
Comment (cameronmeissner) [reply parts/linux/cloud-init/artifacts/cse_config.sh]: moved

**Review by lilypan26 — APPROVED** [review]

File: staging/cse/windows/kubeletfunc.ps1 (hunk 1 above)
Comment (cameronmeissner) [reply staging/cse/windows/kubeletfunc.ps1:224]: added

**Pull request merged**